march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -317,33 +317,35 @@ void MarchHardwareInterface::write(const ros::Duration& elapsed_time)
   }
 }
 
-void MarchHardwareInterface::resetIMotionCubesUntilTheyWork()
+void MarchHardwareInterface::initiateIMC()
 {
-  bool encoderSetCorrectly = false;
-
-  while (!encoderSetCorrectly)
+  for (int i = 0; i < num_joints_; ++i)
   {
-    encoderSetCorrectly = true;
-    for (int i = 0; i < num_joints_; ++i)
+    march4cpp::Joint joint = marchRobot.getJoint(joint_names_[i]);
+    if (joint.getAngleIU() == 0)
     {
-      march4cpp::Joint joint = marchRobot.getJoint(joint_names_[i]);
-      if (joint.getAngleIU() == 0)
-      {
-        ROS_ERROR("[%s] Failed (encoder reset)", joint_names_[i].c_str());
-        encoderSetCorrectly = false;
-      }
+      ROS_WARN_THROTTLE(1, "[%s] has encoder value 0 ", joint_names_[i].c_str());
```

Comment:
I don't think we want to throttle this, since it should be printed for every joint that has this issue. I think this whole loop happens within 1 second, so it will always print 0 or 1 joint.
```suggestion
      ROS_WARN("[%s] has encoder value 0 ", joint_names_[i].c_str());
```